### PR TITLE
add TLS and non-TLS tests

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

File: functional/FunctionalTest.java [hunk 1]
```diff
@@ -3880,7 +3909,6 @@ public static void main(String[] args) throws Exception {
       region = args[3];
     }
 
-    isSecureEndpoint = endpoint.toLowerCase(Locale.US).contains("https://");
```
Comment (balamurugana) [inline functional/FunctionalTest.java:3883]: If `endpoint` is a passed argument, we should have this.

File: functional/FunctionalTest.java [hunk 2]
```diff
@@ -3823,12 +3836,112 @@ public static Process runMinio() throws Exception {
...
+              .endpoint(endpoint)
+              .credentials(accessKey, secretKey)
+              .region(region)
+              .build();
+      adminClientTests = new TestMinioAdminClient(adminClient, mintEnv);
+      FunctionalTest.runTests();
+    }
+  }
+
+  public static void runCustomEndpointTests() throws Exception {
```
Comment (balamurugana) [inline functional/FunctionalTest.java:3903]: This method is exactly duplicate of above with endpoint change. We should have one test runner irrespective of endpoint.

File: functional/FunctionalTest.java [hunk 3]
```diff
@@ -3823,12 +3836,112 @@ public static Process runMinio() throws Exception {
...
+    isSecureEndpoint = endpoint.toLowerCase(Locale.US).contains("https://");
+    client = MinioClient.builder().endpoint(endpoint).credentials(accessKey, secretKey).build();
+    client.ignoreCertCheck();
```
Comment (balamurugana) [inline functional/FunctionalTest.java:3906]: No need to do this for endpoint argument passed. User knows how to use functional test in this case.

File: functional/FunctionalTest.java (hunk 1 above)
Comment (balamurugana) [reply functional/FunctionalTest.java:3883]: The actual fix is not removing this line and ignore using `endpointTLS` down for this case.

File: .github/workflows/gradle.yml [hunk 4]
```diff
@@ -35,6 +35,7 @@ jobs:
           ./gradlew runFunctionalTest
           javac -cp api/build/libs/minio-${DEV_VERSION}-all.jar functional/TestUserAgent.java
           java -Dversion=${DEV_VERSION} -cp api/build/libs/minio-${DEV_VERSION}-all.jar:functional TestUserAgent
+          ./gradlew clean
```
Comment (balamurugana) [inline .github/workflows/gradle.yml:38]: What problem did you see to add this line?

Comment (harshavardhana) [reply .github/workflows/gradle.yml:38]: This is mainly a problem that happens sporadically causing TestUserAgent.java build to fail.

File: functional/FunctionalTest.java [hunk 5]
```diff
@@ -3823,12 +3836,70 @@ public static Process runMinio() throws Exception {
...
+    if (automated) {
+      // Run TLS tests on TLS endpoint
+      client =
+          MinioClient.builder().endpoint(endpointTLS).credentials(accessKey, secretKey).build();
+      client.ignoreCertCheck();
+      adminClient =
+          MinioAdminClient.builder()
+              .endpoint(endpointTLS)
+              .credentials(accessKey, secretKey)
+              .httpClient(getUnsafeOkHttpClient())
```
Comment (balamurugana) [inline functional/FunctionalTest.java:3868]: You could remove this and use `adminClient.ignoreCertCheck()`

**Review by balamurugana — COMMENTED** [review]
Minor correction

File: functional/FunctionalTest.java (hunk 5 above)
Comment (harshavardhana) [reply functional/FunctionalTest.java:3868]: That doesn't work @balamurugana

Comment (balamurugana) [reply functional/FunctionalTest.java:3868]: It seems `MinioAdminClient` has implementation problem. I will fix the issue in different PR

**Review by balamurugana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**